### DISCUSSION TITLE
How do I edit CS in the demo

### DISCUSSION THREAD

**Discussion by martynia** [DIRACGrid/diracx#244] [Developer discussions]
Hi,
As discussed during the last Dirac meeting I would like moo be able to edit the CS to be used with a demo session. How can it be done ?
 Thanks, JM

**Comment by chaen** [comment]
Here
https://github.com/DIRACGrid/diracx-charts/blob/master/docs/RUN_DEMO.md#diracx-configuration

**Reply by martynia** [reply]
Could you explain what _"this file"_ in the section above refers to ? the `default.yml` file is generated by `run_demo.sh `adding some bits to it, but where are the other sections of the file taken from ?

**Reply by chaen** [reply]
what other sections ?

**Comment by martynia** [comment]
Lets me re-phrase: 

Small steps:How do I add another property to DiracAdmin user ? At the moment it is hardwired in the code to be "NormalUser". 
This is **the only** allowed property *) for this user at the moment. How do I add another property i.e. _Operator_ to this user ?

From a description you are linking above there is a mysterious ("this") file to be modified and committed. Which file ?

*` dirac login --property NormalUser diracAdmin` is the only combination which works. Replacing` NormalUser` for `Operator` fails (and `NormalUser` is the (only) default property so it may be omitted.)

**Reply by chaen** [reply]
Ah okay, so having context really helps.
Few points here:
* the `--property` flag is just a place holder and does nothing yet (I agree, it's misleading) https://github.com/DIRACGrid/diracx/issues/148
* The property of the group we add is hard coded at the moment: https://github.com/DIRACGrid/diracx/blob/2e180849611f4628d4fdaf2a663e1c7b5be94b2d/diracx-cli/src/diracx/cli/internal/__init__.py#L75   (you could add a extra function to add a property to a group)
* The mysterious file (default.yaml) contains the list of properties for the group you are interested in. Just add it, and commit

**Comment by martynia** [comment]
OK,
Am I supposed to add a mounted default.yml?
It’s in a ignored directory, should I mess up with that?
J.

**Reply by chaen** [reply]
Sorry but I don't understand.
Don't you have a file `diracx-charts/.demo/cs-mount/initialRepo/default.yml` ?

**Reply by martynia** [reply]
Yes I do have this file.  The file in not on GitHub ... because:

as I pointed out above, the file is in `.demo` which is in `.gitignore`. As a consequence the `.demo` directory never makes it to the _charts_ repository. I assume it is an intended behaviour. This file is not present in a `diracx` repo either(as far as I can see - maybe there is a template somewhere ...)

 I _could_ in principle modify `.gitignore`  and commit that file to _charts_ . Is this what you want me to do ?

A side note: according to git docs, one cannot "unignore" a file (here `diracx-charts/.demo/cs-mount/initialRepo/default.yml` from an ignored directory (here `.demo)`. An exclamation mark (!<file_to_white_list>) does not work.

This is cumbersome since I would have to remove that commit on every push, again assuming that the `.demo` directory is excluded for a reason.

**Reply by chaen** [reply]
The `.demo` directory is created on the fly, and should by no mean be committed to the repo. The `diracx-charts/.demo/cs-mount/initialRepo` is a git repository on its own. 

Your question is: 
> How do I edit CS in the demo

The answer is
> Edit that file `default.yml` file, and commit in the `initialRepo`

If you want your CS changes to be an actual part of the whole demo installation such that it is always there, then you need to add an extra function (see for example `add_vo` in the cli) to add the property.

I think that we would all save a lot of time if you could give a clearer description of what you are trying to do from the beginning :-)

**Reply by martynia** [reply]
> Edit that file `default.yml` file, and commit in the `initialRepo`

Thanks. It is a golden statement ;-)
